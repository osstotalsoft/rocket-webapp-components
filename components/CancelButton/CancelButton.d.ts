import { ButtonProps } from '@material-ui/core/Button';

export interface CancelButtonProps extends Omit<ButtonProps, 'color'> {
    /**
    * The title of button.
    */
    title?: string
    /**
     * Callback fired when a "click" event is detected.
     */
    onClick?: () => void
    /**
    * If true, the button will be disabled.
    */
    disabled?: boolean
    /**
    * @default "theme"
    * The color of button.
    */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground"
    /**
    * @default "medium "
    * The size of button.
    */
    size?: "medium" | "small"
    /**
    *@default "small"
    * The fontSize of button's content.
    */
    fontSize?: "inherit" | "medium" | "small" | "large"
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/cancel-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/CancelButton/CancelButton.js
 * 
 * Material-UI components used:
 * 
 * - IconButton: https://material-ui.com/api/icon-button/
 * 
 */
export default function CancelButton(props: CancelButtonProps): JSX.Element;