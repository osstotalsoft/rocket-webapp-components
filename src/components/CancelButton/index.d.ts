export { default } from './CancelButton';
export * from './CancelButton';