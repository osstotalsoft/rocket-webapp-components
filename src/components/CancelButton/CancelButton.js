import React from "react";
import PropTypes from "prop-types";
import CancelIcon from '@material-ui/icons/Cancel';
import CustomIconButton from '../IconButton';

const CancelButton = ({ title, onClick, disabled, fontSize, ...rest }) => (
    <CustomIconButton aria-label="Cancel" onClick={onClick} disabled={disabled} tooltip={title} {...rest}>
        <CancelIcon fontSize={fontSize} />
    </CustomIconButton>
)

CancelButton.defaultProps = {
    color: "theme",
    fontSize: "small",
    size: "medium"
};

CancelButton.propTypes = {
    /**
    * The title of button.
    */
    title: PropTypes.string,
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick: PropTypes.func,
    /**
    * If true, the button will be disabled.
    */
    disabled: PropTypes.bool,
    /**
    * The color of button.
    */
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground",
        "themeWithGradient"
    ]),
    /**
     * The size of button.
     */
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    /**
    * The fontSize of button's content.
    */
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ])
};

export default CancelButton;