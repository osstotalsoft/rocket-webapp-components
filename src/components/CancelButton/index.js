import CancelButton from "./CancelButton";
export default CancelButton;