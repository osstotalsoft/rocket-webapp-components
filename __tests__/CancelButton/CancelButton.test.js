import { mount } from 'enzyme';
import React from 'react';
import CancelButton from '../../components/CancelButton/CancelButton';

describe('CancelButton', () => {
    it('the color, fontSize and size props should be equal to the default values if they are not provided', () => {

        const wrapper = mount(<CancelButton
            title="Cancel"
        ></CancelButton>);

        expect(wrapper.props().color).toBe('theme');
        expect(wrapper.props().fontSize).toBe('small');
        expect(wrapper.props().size).toBe('medium');
    });
});