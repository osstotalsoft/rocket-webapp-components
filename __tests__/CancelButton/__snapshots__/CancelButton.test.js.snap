// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Cancel"
>
  <button
    aria-label="Cancel"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</span>
`;
